### PR TITLE
fix: add missing publishedAt to YouTubePost model

### PR DESCRIPTION
**Pull request by omBratteng**
The model was missing `publishedAt` field, so I added it.
Plus added a real life content test to see if there were any other fields missing.

### CODE REVIEW

File: __tests__/workers/postUpdated.ts [hunk 1]
```diff
@@ -858,6 +876,63 @@ describe('on youtube post', () => {
...
+        like_count: 13,
+        questions: [
+          'What is daily.dev Search?',
+          'How does search recommendations work on daily.dev?',
+          'What are the benefits of using daily.dev Search?',
+        ],
+        summary:
+          'Introducing daily.dev Search, a feature that allows users to dive deeper into topics they have read about on daily.dev. With search recommendations, users can easily find relevant content in their feeds.',
+        view_count: 134,
+        // TODO: check why this was not included in the message from yggdrasil
```
Comment (rebelchris) [inline __tests__/workers/postUpdated.ts:912]: I think we can remove this from api the todo i mean

File: src/entity/posts/YouTubePost.ts [hunk 2]
```diff
@@ -3,6 +3,9 @@ import { Post, PostType } from './Post';
 
 @ChildEntity(PostType.VideoYouTube)
 export class YouTubePost extends Post {
+  @Column({ nullable: true })
+  publishedAt?: Date;
```
Comment (rebelchris) [inline src/entity/posts/YouTubePost.ts:7]: Do we use it anywhere?
I guess its an existing post type?

File: __tests__/workers/postUpdated.ts (hunk 1 above)
Comment (omBratteng) [reply __tests__/workers/postUpdated.ts:912]:
```suggestion
```

Yeah, added a comment on Slack asking about why it wasn't included in the payload.

File: src/entity/posts/YouTubePost.ts (hunk 2 above)
Comment (omBratteng) [reply src/entity/posts/YouTubePost.ts:7]: Not sure, I see it's on the ArticlePost entity, and used in the rss route. Other than that I'm not sure.
Maybe @idoshamun knows

And the worker fails to save the updated content without it.

<details>
<summary>Log entry</summary>

```json
{
  "insertId": "9lzqwny928llcxge",
  "jsonPayload": {
    "trace_flags": "00",
    "span_id": "8373d6afc5bb2d02",
    "messageId": "9841478423361319",
    "level": 50,
    "err": {
      "type": "EntityPropertyNotFoundError",
      "message": "Property \"publishedAt\" was not found in \"YouTubePost\". Make sure your query is correct.",
      "stack": "EntityPropertyNotFoundError: Property \"publishedAt\" was not found in \"YouTubePost\". Make sure your query is correct.\n    at /opt/app/node_modules/typeorm/query-builder/UpdateQueryBuilder.js:307:27\n    at Array.forEach (<anonymous>)\n    at UpdateQueryBuilder.createUpdateExpression (/opt/app/node_modules/typeorm/query-builder/UpdateQueryBuilder.js:303:68)\n    at UpdateQueryBuilder.getQuery (/opt/app/node_modules/typeorm/query-builder/UpdateQueryBuilder.js:34:21)\n    at UpdateQueryBuilder.getQueryAndParameters (/opt/app/node_modules/typeorm/query-builder/QueryBuilder.js:274:28)\n    at UpdateQueryBuilder.execute (/opt/app/node_modules/typeorm/query-builder/UpdateQueryBuilder.js:81:50)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async updatePost (/opt/app/src/workers/postUpdated.js:199:5)\n    at async /opt/app/src/workers/postUpdated.js:351:21\n    at async EntityManager.transaction (/opt/app/node_modules/typeorm/entity-manager/EntityManager.js:73:28)"
    },
    "data": {
      "updated_at": "2023-12-11T13:28:36.997703Z",
      "origin": "squads",
      "extra": {
        "channel_title": "daily dev",
        "description": "Try it out: https://daily.dev/daily-dev-search",
        "comment_count": 3,
        "duration": 63,
        "like_count": 13,
        "content_curation": [
          "release"
        ],
        "questions": [
          "What is daily.dev Search?",
          "How does search recommendations work on daily.dev?",
          "What are the benefits of using daily.dev Search?"
        ],
        "keywords": [
          "developer-tools",
          "search-recommendations",
          "daily-dev-search"
        ],
        "summary": "Introducing daily.dev Search, a feature that allows users to dive deeper into topics they have read about on daily.dev. With search recommendations, users can easily find relevant content in their feeds.",
        "view_count": 134
      },
      "order": 0,
      "image": "https://i.ytimg.com/vi/T_AbQGe7fuU/sddefault.jpg",
      "source_id": "unknown",
      "url": "https://www.youtube.com/watch?v=T_AbQGe7fuU",
      "published_at": "2023-11-07T12:04:12Z",
      "title": "Introducing daily.dev Search",
      "content_type": "video:youtube",
      "post_id": "HR6jmCxzE",
      "id": "7922f432-f554-5967-80b5-932fe7320ac2"
    },
    "subscription": "api.content-published",
    "pid": 8,
    "msg": "failed to update post",
    "trace_id": "8252ba303a1cc4e713564156c7d58fdd",
    "hostname": "api-bg-5f8f4bffb6-pdxkw"
  },
  "resource": {
    "type": "k8s_container",
    "labels": {
      "pod_name": "api-bg-5f8f4bffb6-pdxkw",
      "location": "us-central1",
      "project_id": "devkit-prod",
      "container_name": "app",
      "cluster_name": "prod-vpc-native",
      "namespace_name": "daily"
    }
  },
  "timestamp": "2023-12-12T19:15:38.980Z",
  "labels": {
    "k8s-pod/app": "api-bg",
    "compute.googleapis.com/resource_name": "gke-prod-vpc-native-default-pool-9ca867f7-i83f",
    "k8s-pod/pod-template-hash": "5f8f4bffb6"
  },
  "logName": "projects/devkit-prod/logs/stdout",
  "receiveTimestamp": "2023-12-12T19:15:43.772312500Z",
  "errorGroups": [
    {
      "id": "COOkyrnQpOzPgQE"
    }
  ]
}
```

</details>

Comment (idoshamun) [reply src/entity/posts/YouTubePost.ts:7]: We don't use it anywhere but track it. We can drop it if we want down the line

**Review by idoshamun — APPROVED** [review]

**Pull request merged**